File: rust/src/models/token_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Token response as per OAuth 2.0 spec
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct TokenResponse {
    pub access_token: String,

    pub expires_in: i32,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub refresh_token: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub scope: Option<String>,

    pub token_type: String,
}

impl TokenResponse {
    pub fn new(access_token: String, expires_in: i32, token_type: String) -> Self {
        Self {
            access_token,
            expires_in,
            refresh_token: None,
            scope: None,
            token_type,
        }
    }
}

File: typescript/src/models/tokenResponse.ts
// this file is @generated
/** Token response as per OAuth 2.0 spec */
export interface TokenResponse {
  accessToken: string;

  expiresIn: number;

  refreshToken?: string | null;

  scope?: string | null;

  tokenType: string;
}

export const TokenResponseSerializer = {
  _fromJsonObject(object: any): TokenResponse {
    return {
      accessToken: object["access_token"],
      expiresIn: object["expires_in"],
      refreshToken: object["refresh_token"],
      scope: object["scope"],
      tokenType: object["token_type"],
    };
  },

  _toJsonObject(self: TokenResponse): any {
    return {
      access_token: self.accessToken,
      expires_in: self.expiresIn,
      refresh_token: self.refreshToken,
      scope: self.scope,
      token_type: self.tokenType,
    };
  },
};
